clients/spark/core/src/main/scala/io/treeverse/clients/StorageUtils.scala

```diff
@@ -80,5 +87,71 @@ object StorageUtils {
       if (keys.isEmpty) return Seq.empty
       keys.map(x => snPrefix.concat(x))
     }
+
+    def createAndValidateS3Client(
+        configuration: ClientConfiguration,
+        credentialsProvider: Option[AWSCredentialsProvider],
+        awsS3ClientBuilder: AmazonS3ClientBuilder,
+        region: String,
+        bucket: String
+    ): AmazonS3 = {
+
```

Comment:
Nit: no blank lines at tops and bottoms of functions.  (No, we do not have a standard for this, I just like to squash more lines onto my screen...)

```suggestion
```